### PR TITLE
Saving and Loading for activity based methodologies

### PR DESCRIPTION
**Pull request by isaaccodekill**
Depends on this pr: 

https://github.com/user-attachments/assets/f08fac07-c940-4484-8cf2-d480ba5432dc

https://github.com/Open-Earth-Foundation/CityCatalyst/pull/684

### CODE REVIEW

File: .DS_Store
Comment (lemilonkh) [inline .DS_Store:1]: Delete this file.

File: app/package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "ai": "^3.2.35",
     "autoprefixer": "10.4.16",
     "bcrypt": "^5.1.1",
+    "chakra-react-select": "^4.9.1",
```
Comment (lemilonkh) [inline app/package.json:59]: We had a component like this before on the file upload form. Can you check if you can reuse the same thing as there instead of introducing this dependency?

File: app/src/components/Modals/activity-modal/activity-form-modal.tsx [hunk 2]
```diff
@@ -182,21 +157,43 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
...
     const gasValues = extractGasesAndUnits(activity);
+
+    // extract field values
+    const values: Record<string, any> = {};
+    fields?.forEach((field) => {
+      if (field.id in activity) {
+        values[field.id] = (activity as any)[field.id];
+      }
+      if (field.units) {
+        values[`${field.id}unit`] = (activity as any)[`${field.id}unit`];
```
Comment (lemilonkh) [inline app/src/components/Modals/activity-modal/activity-form-modal.tsx:168]:
```suggestion
        values[`${field.id}Unit`] = (activity as any)[`${field.id}Unit`];
```
Would be nicer for readability, so it's properly camelCased.

File: app/src/components/Modals/activity-modal/activity-form-modal.tsx [hunk 3]
```diff
@@ -182,21 +157,43 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
...
-      inventoryValue: {
-        inputMethodology: getInputMethodology(methodology?.id), // extract methodology name
-        gpcReferenceNumber: referenceNumber,
-        unavailableReason: "",
-        unavailableExplanation: "",
+      activityData: methodology?.id.includes("direct-measure")
+        ? {
+            co2_amount: gasValues[1].factor,
+            ch4_amount: gasValues[0].factor,
+            n2o_amount: gasValues[2].factor,
```
Comment (lemilonkh) [inline app/src/components/Modals/activity-modal/activity-form-modal.tsx:178]: Emissions factors are not direct measure gas values, see feedback on #648

File: app/src/components/Modals/activity-modal/activity-modal-body.tsx [hunk 4]
```diff
@@ -570,7 +723,7 @@ const ActivityModalBody = ({
               borderColor: "content.link",
             }}
             placeholder={t("source-reference-placeholder")}
-            {...register("activity.sourceReference", {
+            {...register(`activity.sourceReference` as any, {
```
Comment (lemilonkh) [inline app/src/components/Modals/activity-modal/activity-modal-body.tsx:726]: Why as any? This takes a string.

File: app/tests/api/activity_value.test.ts [hunk 5]
```diff
@@ -478,6 +486,5 @@ describe("Activity Value API", () => {
 
     const { data } = await res3.json();
     assert.equal(res3.status, 200);
-    assert.equal(data.deletedCount, 3);
```
Comment (lemilonkh) [inline app/tests/api/activity_value.test.ts:481]: Why did you need to remove this assertion? It should still be correct, right?

**Review by lemilonkh — APPROVED** [review]

File: app/package.json (hunk 1 above)
Comment (isaaccodekill) [reply app/package.json:59]: This has been changed in a follow up pr

File: app/src/components/Modals/activity-modal/activity-form-modal.tsx (hunk 3 above)
Comment (isaaccodekill) [reply app/src/components/Modals/activity-modal/activity-form-modal.tsx:178]: Created a PR for this

File: app/src/components/Modals/activity-modal/activity-form-modal.tsx (hunk 2 above)
Comment (isaaccodekill) [reply app/src/components/Modals/activity-modal/activity-form-modal.tsx]: So this might be a bit complex to change because it refers to a field in the form reused in multiple places, so I'll improve this in the next pr

File: app/tests/api/activity_value.test.ts (hunk 5 above)
Comment (isaaccodekill) [reply app/tests/api/activity_value.test.ts]: This tests has to be re-modelled, I plan to rewrite them as they are flaky
